### PR TITLE
feature/query and update animals table

### PR DESCRIPTION
**Pull request by lucy-sees**
### - In this part of the project, I did the following,

- [x] https://github.com/lucy-sees/VeterinaryVentures/pull/2  (previous commits on these tasks)
- [x] Added a column `species` of type string to the `animals` table. Modified the schema.sql file.
- [x] Inserted some data into the table.
- [x] Added these queries:

Inside a transaction update the `animals` table by setting the species column to unspecified. Verify that the change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
screenshot-4

### - Inside a transaction I,

- [x] Updated the `animals` table by setting the species column to `digimon` for all animals that have a name ending in `mon`.
- [x] Updated the `animals` table by setting the species column to `pokemon` for all animals that don't have species already set.
- [x] Verified that change was made and persists after commit.

screenshot-5

### -  Inside a transaction I 
- [x] deleted all records in the `animals` table, then rolled back the transaction.
- [x] After the rollback I verified if all records in the `animals` table still exist.

screenshot-6

### - Inside a transaction, I:

- [x] Deleted all animals born after Jan 1st, 2022.
- [x] Created a savepoint for the transaction.
- [x] Updated all animals' weight to be their weight multiplied by -1.
- [x] Rolledback to the savepoint
- [x] Updated all animals' weights that are negative to be their weight multiplied by -1.
- [x] Committed transaction

screenshot-7

### - Wrote queries to answer the following questions:
- [x] How many animals are there?
- [x] How many animals have never tried to escape?
- [x] What is the average weight of animals?
- [x] Who escapes the most, neutered or not neutered animals?
- [x] What is the minimum and maximum weight of each type of animal?
- [x] What is the average number of escape attempts per animal type of those born between 1990 and 2000?

screenshot-8

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -16,3 +16,32 @@ UPDATE animals SET species = 'unspecified';
...
+BEGIN;
+DELETE  FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO my_savepoint;
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals; -- Verify that the changes were made
+
+SELECT COUNT(*) FROM animals;
```
Comment (OyePriscilla) [inline queries.sql:42]: - [ ] Please kindly include the screenshot of this query to the PR as well as stated in the requirement. 🙏 

Untitled

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @lucy-sees 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implemented transactions to update and delete records ✔️ 

## Required Changes ♻️

- [ ] Kindly fix the broken license link in the README file, I got 404 error. Implementing this will make your repo to be more professional. 👍 

## Optional suggestions

- [ ] KIndly ensure that all the project requirement for a particular milestone are implemented in the feature branch. 👍 I observed that you had already done part of the requirement in the dev branch, this will not allow me to be add inline-comment if need be. 🙏 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lucy-sees** [comment]
### - Wrote queries to answer the following questions:
- [x] How many animals are there?
- [x] How many animals have never tried to escape?
- [x] What is the average weight of animals?
- [x] Who escapes the most, neutered or not neutered animals?
- [x] What is the minimum and maximum weight of each type of animal?
- [x] What is the average number of escape attempts per animal type of those born between 1990 and 2000?

screenshot-8

**Review by Strangeal — APPROVED** [review]
Hi @lucy-sees, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Added screenshot for all queries ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**